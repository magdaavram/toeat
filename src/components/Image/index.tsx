import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';
import Breakfast from './images/Breakfast.png';
import Brunch from './images/Brunch.png';
import Dinner from './images/Dinner.png';
import Lunch from './images/Lunch.png';
import Snack from './images/Snack.png';

interface IImageProps extends InputHTMLAttributes<HTMLDivElement> {
  url: string;
  course: string;
}

const ImageStyle = styled.div(({ url }: IImageProps) => ({
  width: '400px',
  height: '360px',
  background: `url(${url}) no-repeat center`,
  backgroundSize: 'cover',
}));

const Image = ({ url, course, className }: IImageProps) => {
  return <ImageStyle className={className} url={getUrl(url, course)} course={course} />;
};

export default Image;

const getUrl = (url: string, course: string): string => {
  let imageUrl = url;

  if (url === '') {
    const defaultImages: any = {
      Breakfast: Breakfast,
      Brunch: Brunch,
      Dinner: Dinner,
      Lunch: Lunch,
      Snack: Snack,
    };

    imageUrl = defaultImages[course] || Lunch;
  }

  return imageUrl;
};
