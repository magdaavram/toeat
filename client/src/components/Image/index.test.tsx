import React from 'react';
import {mount} from 'enzyme';
import Image from './index';

describe('Image component should render correctly', () => {
  const tests = [
    {
      description: 'should render given image',
      props: {url: 'image.png', course: 'Breakfast'}
    },
    {
      description: 'should render default image (breakfast)',
      props: {course: 'Breakfast'}
    },
    {
      description: 'should render default image (brunch)',
      props: {course: 'Brunch'}
    },
    {
      description: 'should render default image (dinner)',
      props: {course: 'Dinner'}
    },
    {
      description: 'should render default image (lunch)',
      props: {course: 'Lunch'}
    },
    {
      description: 'should render default image (snack)',
      props: {course: 'Snack'}
    },
    {
      description: 'should render default image (course not declared)',
      props: {course: 'Supper'}
    },
  ];

  tests.forEach(test => {
    it(test.description, () => {
      const wrapper = mount(<Image {...test.props}/>);

      expect(wrapper).toMatchSnapshot();
    });
  })
});
