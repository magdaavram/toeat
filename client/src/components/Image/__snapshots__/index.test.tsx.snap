// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly should render default image (breakfast) 1`] = `
<Image
  course="Breakfast"
>
  <styled.div
    course="Breakfast"
    url="Breakfast.png"
  >
    <div
      className="sc-AxjAm cOFEbK"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render default image (brunch) 1`] = `
<Image
  course="Brunch"
>
  <styled.div
    course="Brunch"
    url="Brunch.png"
  >
    <div
      className="sc-AxjAm iZspXd"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render default image (course not declared) 1`] = `
<Image
  course="Supper"
>
  <styled.div
    course="Supper"
    url="Lunch.png"
  >
    <div
      className="sc-AxjAm gtwHQd"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render default image (dinner) 1`] = `
<Image
  course="Dinner"
>
  <styled.div
    course="Dinner"
    url="Dinner.png"
  >
    <div
      className="sc-AxjAm lnDKUf"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render default image (lunch) 1`] = `
<Image
  course="Lunch"
>
  <styled.div
    course="Lunch"
    url="Lunch.png"
  >
    <div
      className="sc-AxjAm gtwHQd"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render default image (snack) 1`] = `
<Image
  course="Snack"
>
  <styled.div
    course="Snack"
    url="Snack.png"
  >
    <div
      className="sc-AxjAm lbdUsR"
    />
  </styled.div>
</Image>
`;

exports[`<Image /> should render correctly should render given image 1`] = `
<Image
  course="Breakfast"
  url="image.png"
>
  <styled.div
    course="Breakfast"
    url="image.png"
  >
    <div
      className="sc-AxjAm gXdLeG"
    />
  </styled.div>
</Image>
`;
